### PR TITLE
feat(core): add ability to align dropdown to center

### PR DESCRIPTION
**Pull request by ikurilov**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Does not align dropdown to center

Closes # [3704](https://github.com/Tinkoff/taiga-ui/issues/3704)

## What is the new behavior?

Aligns dropdown to center

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/core/directives/dropdown/dropdown-position.directive.ts [hunk 1]
```diff
@@ -44,6 +44,10 @@ export class TuiDropdownPositionDirective extends TuiPositionAccessor {
             top: hostRect.top - offset - height,
             bottom: hostRect.bottom + offset,
             right,
+            center:
+                hostRect.left + width < viewport.right - offset
```
Comment (waterplea) [inline projects/core/directives/dropdown/dropdown-position.directive.ts:48]: Shouldn't it be `hostRect.left + hostRect.width / 2 + width / 2 < viewport.right - offset`?

File: projects/core/directives/dropdown/dropdown-position-sided.directive.ts [hunk 2]
```diff
@@ -46,7 +46,8 @@ export class TuiDropdownPositionSidedDirective extends TuiPositionAccessor {
         const {height, width} = rect;
         const hostRect = this.vertical.accessor?.getClientRect() ?? EMPTY_CLIENT_RECT;
         const viewport = this.viewport.getClientRect();
-        const {align, direction, minHeight, offset} = this.options;
+        const {direction, minHeight, offset} = this.options;
+        const align = this.options.align === 'center' ? 'left' : this.options.align;
```
Comment (waterplea) [inline projects/core/directives/dropdown/dropdown-position-sided.directive.ts:50]: I imagine people would want sided center too 🤔 But we can probably leave it until somebody actually asks for it.

**Review by waterplea — APPROVED** [review]

File: projects/core/directives/dropdown/dropdown-position.directive.ts (hunk 1 above)
Comment (ikurilov) [reply projects/core/directives/dropdown/dropdown-position.directive.ts:48]: yes, I guess so

File: projects/core/directives/dropdown/dropdown-position-sided.directive.ts (hunk 2 above)
Comment (ikurilov) [reply projects/core/directives/dropdown/dropdown-position-sided.directive.ts:50]: I thought about it for a while and tuiDropdownAlign = 'center' in sided dropdown seems strange. May be tuiDropdownDirection = 'center' would be better for it, but will look strange for default dropdowns...

**Pull request merged**